packages/poaps/src/PoapsClient.ts

```diff
@@ -90,4 +108,145 @@ export class PoapsClient {
       nextCursor(poaps.length, limit, offset),
     );
   }
+
+  /**
+   * Retrieves the secret code associated with a POAP code.
+   * @async
+   * @param {string} mintCode - The POAP code for which to get the secret.
+   * @returns {Promise<string>} The associated secret code.
+   * @throws {CodeAlreadyMintedError} Thrown when the POAP code has already been minted.
+   * @throws {CodeExpiredError} Thrown when the POAP code is expired.
+   */
+  private async getSecretCode(mintCode: string): Promise<string> {
+    const getCodeResponse = await this.getMintCode(mintCode);
+
+    if (getCodeResponse.claimed == true) {
+      throw new CodeAlreadyMintedError(mintCode);
+    }
+    if (getCodeResponse.is_active == false) {
+      throw new CodeExpiredError(mintCode);
+    }
+
+    return getCodeResponse.secret;
+  }
+
+  /**
+   * Retrieves mint code details for a specific QR hash.
+   * @async
+   * @param {string} mintCode - The QR hash for which to get the mint code.
+   * @returns {Promise<GetMintCodeResponse>} The mint code details.
+   */
+  async getMintCode(mintCode: string): Promise<GetMintCodeResponse> {
+    return await this.tokensApiProvider.getMintCode(mintCode);
+  }
+
+  /**
+   * Fetches the current status of a mint based on its unique ID.
+   * @async
+   * @param {string} queueUid - The unique ID of the mint.
+   * @returns {Promise<MintingStatus>} The current status of the mint.
+   */
+  async getMintStatus(queueUid: string): Promise<MintingStatus> {
+    const mintStatusResponse = await this.tokensApiProvider.mintStatus(
+      queueUid,
+    );
+    return mintStatusResponse.status;
+  }
+
+  /**
+   * Awaits until the mint's status changes from 'IN_PROCESS' or 'PENDING'.
+   * @async
+   * @param {string} queueUid - The unique ID of the mint.
+   * @returns {Promise<void>}
+   */
+  async waitMintStatus(queueUid: string, mintCode: string): Promise<void> {
+    const checker = new MintChecker(queueUid, this.tokensApiProvider, mintCode);
+    await checker.checkMintStatus();
+  }
+
+  /**
+   * Awaits until a specific POAP, identified by its QR hash, is indexed.
+   * @async
+   * @param {string} mintCode - The QR hash identifying the POAP to be indexed.
+   * @returns {Promise<GetMintCodeResponse>} Details of the indexed POAP.
+   */
+  async waitPoapIndexed(mintCode: string): Promise<GetMintCodeResponse> {
+    const checker = new PoapIndexed(mintCode, this.tokensApiProvider);
+    return await checker.waitPoapIndexed();
+  }
```

Comment:
In this function you cannot return `GetMintCodeResponse` because it's the RAW response from Compass and is snake_case. You need to wrap it in DTO as you do with all objects returned by the SDK.